core/cloudflow-operator/src/main/scala/cloudflow/operator/CloudflowApplication.scala

```diff
@@ -186,42 +244,89 @@ object CloudflowApplication {
-        }
-        .getOrElse(PodStatus(name, "Unknown", 0, ""))
+      // See https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/
+      val name                = pod.metadata.name
+      val status              = pod.status.getOrElse(Pod.Status())
+      val nrOfContainers      = pod.spec.map(_.containers.size).getOrElse(status.containerStatuses.size)
+      val nrOfContainersReady = status.containerStatuses.filter(_.ready == true).size
+      val restarts            = status.containerStatuses.map(_.restartCount).sum
+      val containerStates     = status.containerStatuses.flatMap(_.state)
+
+      // https://github.com/kubernetes/kubectl/blob/27fa797464ff6684ec591c46c69d5e013998a0d1/pkg/describe/describe.go#L698
+      val st = if (pod.metadata.deletionTimestamp.nonEmpty) {
+        Terminating
+      } else {
+        status.phase
+          .map {
+            case Pod.Phase.Pending   ⇒ getStatusFromContainerStates(containerStates, nrOfContainers)
+            case Pod.Phase.Running   ⇒ Running
+            case Pod.Phase.Succeeded ⇒ Succeeded
+            case Pod.Phase.Failed    ⇒ Failed
+            case Pod.Phase.Unknown   ⇒ Unknown
+          }
+          .getOrElse(getStatusFromContainerStates(containerStates, nrOfContainers))
+      }
+      PodStatus(name, st, restarts, nrOfContainersReady, nrOfContainers)
     }
+
+    private def getStatusFromContainerStates(containerStates: List[Container.State], nrOfContainers: Int) =
+      if (containerStates.nonEmpty) {
+        // - Running if all containers running;
+        // - Terminated if all containers terminated;
+        // - first Waiting reason found (ContainerCreating, CrashLoopBackOff, ErrImagePull, ...);
+        // - otherwise Pending.
+        if (containerStates.collect { case _: Container.Running => 1 }.sum == nrOfContainers) Running
+        else if (containerStates.collect { case _: Container.Terminated => 1 }.sum == nrOfContainers) Terminated
+        else {
+          containerStates
+            .collect { case Container.Waiting(Some(reason)) => reason }
+            .headOption
+            .getOrElse(Pending)
+        }
+      } else Pending
   }
 
   /**
    * Status of the pod.
    * ready can be "True", "False" or "Unknown"
    */
-  case class PodStatus(name: String, status: String, restarts: Int, ready: String)
-
+  case class PodStatus(
+      name: String,
+      status: String,
+      restarts: Int = 0,
+      // Optional for backwards compatibility
+      nrOfContainersReady: Option[Int] = None,
+      // Optional for backwards compatibility
+      nrOfContainers: Option[Int] = None,
+      // TODO can this be removed without breaking backwards-compatibility?
+      ready: Option[String] = None
+  ) {
+    def containers      = nrOfContainers.getOrElse(0)
+    def containersReady = nrOfContainersReady.getOrElse(0)
+    def isReady         = status == PodStatus.Running && containersReady == containers && containers != 0
```

Comment:
```suggestion
    def isReady         = containers > 0 && status == PodStatus.Running && containersReady == containers
```

Reply:
I fixed this too, out of date somehow so could not just commit it from github.